### ISSUE TITLE
RUF009 hints for float('-inf')

### ISSUE THREAD

**Issue by alkatar21** [charliermarsh/ruff#4657]
`RUF009` is hinting in a dataclass where floats are initialized with `float('-inf')`.
As suggested in [Discord](https://discord.com/channels/1039017663004942429/1039017663512449056/1111040962039386122) `float` should probably be added to the list of allowed functions.

```py
from dataclasses import dataclass

@dataclass
class Test:
    example: float = float('-inf') # Do not perform function call `float` in dataclass defaultsRuff(RUF009)
```

**Comment by JonathanPlasse** [comment]
Why not use `math.inf`?

**Comment by charliermarsh** [comment]
Thanks :)

**Comment by JonathanPlasse** [comment]
@alkatar21, Would you like to make a PR for it?

**Comment by alkatar21** [comment]
If you could give me a hint of where to look, I would try.

**Comment by qdegraaf** [comment]
@alkatar21 Quick scan of the rule suggests this is approximately where the check is done: https://github.com/charliermarsh/ruff/blob/741e180e2d30078c59e03a8761d90757964c5cf7/crates/ruff/src/rules/ruff/rules/mutable_defaults_in_dataclass_fields.rs#L202 

I can also open a PR if you prefer.

**Comment by qdegraaf** [comment]
Should the check for float be added to the `IMMUTABLE_FUNCS` const and therefore `is_immutable_func`, which is also used in flake8-bugbear rule `FunctionCallInDefaultArgument`? Or should it be a separate check for `RUF009` only?

[This line]( https://github.com/PyCQA/flake8-bugbear/blob/a309029eed5070938a376af88b0d046e8acda5c9/bugbear.py#LL1575C25-L1575C25) in `flake8-bugbear` seems to suggest it should just be a `RUF009` thing but just wanna make sure before I (or someone else equally unfamiliar) opens a PR.

**Comment by charliermarsh** [comment]
Yeah, I think float and int and perhaps some other builtins (`complex`?) should be added there.

**Comment by qdegraaf** [comment]
I'll open a PR with those in as well. Looking at:
```rust
const IMMUTABLE_FUNCS: &[&[&str]] = &[
    &["", "tuple"],
    &["", "frozenset"],
    &["datetime", "date"],
    &["datetime", "datetime"],
    &["datetime", "timedelta"],
    &["decimal", "Decimal"],
    &["operator", "attrgetter"],
    &["operator", "itemgetter"],
    &["operator", "methodcaller"],
    &["pathlib", "Path"],
    &["types", "MappingProxyType"],
    &["re", "compile"],
];
```
I'll add `int`, `float`, `complex`.

 If it should be: "all functions that return an immutable object" should `str` and `bool` also be added?
